metadata-service/health-servlet/src/main/java/com/datahub/health/controller/HealthCheckController.java

```diff
@@ -33,6 +35,27 @@ public HealthCheckController() {
         this::getElasticHealth, 5, TimeUnit.SECONDS);
   }
 
+  /**
+   * Combined health check endpoint for checking GMS clients.
+   * For now, just checks the health of the ElasticSearch client
+   * @return A ResponseEntity with a Map of String (component name) to ResponseEntity (the health check status of
+   * that component). The status code will be 200 if all components are okay, and 500 if one or more components are not
+   * healthy.
+   */
+  @GetMapping(path = "/combined", produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<Map<String, ResponseEntity<String>>> getCombinedHealthCheck() {
+    Map<String, ResponseEntity<String>> componentHealth = new HashMap<>();
+
+    componentHealth.put("elastic", getElasticHealthWithCache());
```

Comment:
please make the key match `elasticsearch` since there are several other services with `elastic`, such as `elasticache`